strawberryfields/io/xir_io.py

```diff
@@ -0,0 +1,294 @@
+# Copyright 2019-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for loading and saving Strawberry Fields
+:class:`~.Program` objects from/to Blackbird scripts and Strawberry Fields
+code.
+"""
+# pylint: disable=protected-access,too-many-nested-blocks
+from decimal import Decimal
+from typing import Iterable
+
+import numpy as np
+
+import xir
+from xir.program import Declaration
+
+import strawberryfields.program as sfp
+from strawberryfields.tdm.tdmprogram import TDMProgram
+import strawberryfields.parameters as sfpar
+from strawberryfields import ops
+
+
+# pylint: disable=too-many-branches
+def from_xir(xir_prog):
+    """Convert an XIR Program to a Strawberry Fields program.
+
+    Args:
+        xir_prog (xir.Program): the input XIR program object
+
+    Returns:
+        Program: corresponding Strawberry Fields program
+    """
+    # only script-level statements are part of `xir_prog.statements`, which can only have integer
+    # wires, leading to `xir_prog.wires` only containing integer wire labels
+    num_of_modes = int(max(xir_prog.wires or [-1])) + 1
+    name = xir_prog.options.get("name", "xir")
```

Comment:
What do you think about making `name` reserved and making the name more meaningful?
```suggestion
    name = xir_prog.options.get("name", "sf_from_xir")
```

Reply:
What do you mean by making `name` reserved? Just to document it as reserved and make sure it's always a string? 🤔

Reply:
Yes, sorry, I meant renaming it to `_name_`.

Reply:
Alright, I've changed it now. I think your suggestion makes sense.